File: backend/src/models/KnowledgeBase.ts
import {
    Table,
    Column,
    CreatedAt,
    UpdatedAt,
    Model,
    PrimaryKey,
    AutoIncrement,
    AllowNull,
    ForeignKey,
    BelongsTo,
    HasMany,
    DataType
} from "sequelize-typescript";
import Tenant from "./Tenant";
import KnowledgeSource from "./KnowledgeSource";

@Table
class KnowledgeBase extends Model<KnowledgeBase> {
    @PrimaryKey
    @AutoIncrement
    @Column
    id: number;

    @AllowNull(false)
    @Column
    name: string;

    @AllowNull(true)
    @Column
    description: string;

    @ForeignKey(() => Tenant)
    @Column(DataType.UUID)
    tenantId: string;

    @BelongsTo(() => Tenant)
    tenant: Tenant;

    @HasMany(() => KnowledgeSource)
    sources: KnowledgeSource[];

    @CreatedAt
    createdAt: Date;

    @UpdatedAt
    updatedAt: Date;
}

export default KnowledgeBase;

File: backend-go/internal/models/knowledge_base.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type KnowledgeBase struct {
	ID          int                   `gorm:"primaryKey" json:"id"`
	Name        string                `gorm:"not null" json:"name"`
	Description string                `json:"description"`
	TenantID    uuid.UUID             `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	CreatedAt   time.Time             `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt   time.Time             `gorm:"column:updatedAt" json:"updatedAt"`
	Sources     []KnowledgeBaseSource `gorm:"foreignKey:KnowledgeBaseID" json:"sources,omitempty"`
}

func (KnowledgeBase) TableName() string {
	return "KnowledgeBases"
}
